src/axom/sina/core/AdiakWriter.cpp

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ ******************************************************************************
+ *
+ * \file AdiakWriter.cpp
```

Comment:
Note that we have some potentially similar logic in core/Annotations

See: https://github.com/LLNL/axom/blob/develop/src/axom/core/utilities/Annotations.cpp